content/posts/2020-10-10---Wprowadzenie-do-Mybatisa.md

```diff
@@ -0,0 +1,264 @@
+    private String description;
+    private Integer ownerId;
+}
+```
+
+Teraz należy stworzyć interfejs zwany Mapperem. Jest on kluczowym plikiem w tym frameworku, gdyż na jego podstawie Spring stworzy Bean, który będzie się komunikował z bazą danych. Może on przyjmować argumenty, które będą następnie wykorzystywane podczas tworzenia zapytania przy pomocy SQLa.
+
+**BookDaoMapper.java**
+```java
+package pl.akai.bookcrossing.list;
+
+import org.apache.ibatis.annotations.Mapper;
+import org.apache.ibatis.annotations.Param;
+import pl.akai.bookcrossing.model.Book;
+
+import java.util.List;
+
+
+@Mapper
+public interface BookDaoMapper {
+
+    List<Book> getAllBooks();
+
+    Book getBookById(@Param("id") int id);
+
+    void insertBook(@Param("book") Book book);
+}
+
+```
+Teraz należy dopisać odpowiedni kod w SQLu do stowroznych wcześniej funkcji Mappera w pliku XML. 
+
+**BookDaoMapper.xml**
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd" >
+<mapper namespace="pl.akai.bookcrossing.list.BookDaoMapper">
+
+    <resultMap id="baseBookMap" type="pl.akai.bookcrossing.model.Book" >
+        <id column="id" property="id" jdbcType="INTEGER" />
+        <result column="title" property="title" jdbcType="VARCHAR" />
+        <result column="author" property="author" jdbcType="VARCHAR" />
+        <result column="description" property="description" jdbcType="VARCHAR" />
+        <result column="owner_id" property="ownerId" jdbcType="INTEGER" />
+    </resultMap>
+
+    <select id="getAllBooks" resultMap="baseBookMap">
+        select * from tb_book
+    </select>
+
+    <select id="getBookById" resultMap="baseBookMap">
+        select * from tb_book
+        where id = #{id}
+    </select>
+
+    <insert id="insertBook">
+        INSERT INTO tb_book (title, author, description, owner_id)
+        VALUES (#{book.title}, #{book.author}, #{book.description}, #{book.ownerId});
+    </insert>
+    
+</mapper>
+```
+Pierwszą rzeczą, na którą należy zwrócić uwagę jest `namespace="pl.akai.bookcrossing.list.BookDaoMapper"`. Odpowiada on za połączenie pliku XML z odpowiednim interfejsem. 
+Następnie tworzymy resultMap. Dzięki niej MyBatis wie, w jaki sposób ma połączyć ze sobą pola klasy i kolumny otrzymane poprzez zapytanie SQL. `column` odpowiada kolumnie z zapytania, `property` polu w klasie, a `jdbcType` to typ.
+Na samym końcu piszemy swoje zapytania w języku SQL. `id` takiego zapytania odpowiada nazwie metody z interfejsu Mappera, natomiast `resultMap` pozwala na przekształcenie wyniku zapytania na obiekt Javy. Jedną z największych zalet MyBatisa jest możliwość tworzenia dynamicznych zapytań SQLa, czyli na przykład takich, które mają w swoim wnętrzu na przykład instrukcje warunkowe czy pętle. Zainteresowanych zapraszam do dokumentacji, gdzie są przykłady takich zapytań [Zapytania dynamiczne](https://mybatis.org/mybatis-3/dynamic-sql.html)
```

Comment:
```suggestion
Na samym końcu piszemy swoje zapytania w języku SQL. 

- `id` takiego zapytania odpowiada nazwie metody z interfejsu Mappera, 
- `resultMap` pozwala na przekształcenie wyniku zapytania na obiekt Javy. 

Jedną z największych zalet MyBatisa jest możliwość tworzenia dynamicznych zapytań SQLa, czyli na przykład takich, które mają w swoim wnętrzu na przykład instrukcje warunkowe czy pętle. 

Zainteresowanych zapraszam do dokumentacji, gdzie są przykłady takich zapytań [Zapytania dynamiczne](https://mybatis.org/mybatis-3/dynamic-sql.html)
```

I suggest pointing the names you are describing. It will give more transparency.